### PR TITLE
test: don't spam to console when testing download file

### PR DESCRIPTION
**Pull request by matklad**
I wanted to quickly fix this test printing to console, but ended up refactoring code around a bit as well

### CODE REVIEW

File: nearcore/src/download_file.rs [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+        Ok(std::fs::read(path).unwrap())
+    }
+
+    /// Tests writing plain text of varying lengths through [`AutoXzDecoder`].
+    /// Includes test cases where prefix of a XZ header is present at the beginning
+    /// of the stream being written.  That tests the object not being fooled by
+    /// partial prefix.
+    #[test]
+    fn test_auto_xz_decode_plain() {
+        let mut data: [u8; 38] = *b"A quick brow fox jumps over a lazy dog";
```
Comment (mina86) [inline nearcore/src/download_file.rs:321]: While you’re touching that, perhaps s/brow/brown/.

File: nearcore/src/download_file.rs [hunk 2]
```diff
@@ -0,0 +1,369 @@
+use hyper::body::HttpBody;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::path::{Path, PathBuf};
+use tokio::io::AsyncWriteExt;
+
+#[derive(thiserror::Error, Debug)]
+pub enum FileDownloadError {
```
Comment (mina86) [inline nearcore/src/download_file.rs:7]: This and other public symbols can now be `pub (crate)`.

**Review by mina86 — APPROVED** [review]

File: nearcore/src/download_file.rs [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        let tmp_downloaded_file = tempfile::NamedTempFile::new().unwrap();
+        let tmp_downloaded_file_path = tmp_downloaded_file.path();
+
+        download_file(&format!("http://localhost:{}", port), tmp_downloaded_file_path)
+            .await
+            .unwrap();
+
+        let downloaded_file_content = std::fs::read(tmp_downloaded_file_path).unwrap();
+        assert_eq!(downloaded_file_content, [42; 1024].to_vec());
+    }
```
Comment (mina86) [inline nearcore/src/download_file.rs:296]: How about converting it into a helper function with `expected` and `payload` arguments and than running the test once for plain data and another time for XZ-compressed data?

**Review by nikurt — APPROVED** [review]

File: nearcore/src/download_file.rs (hunk 3 above)
Comment (matklad) [reply nearcore/src/download_file.rs:296]: I've thought about this, but I think it's better not to do this. The way tests are written, we have control over decoder's state machine (`for chunk_size` loops). If we hit it purely via downloading, than I think we'll only handle the boring case where all the data is available up-front.

Comment (matklad) [reply nearcore/src/download_file.rs]: After thinking for one more minute, I realized that it's probably better to add the tests you suggest, but keep the old ones as well :-)

**Pull request merged**